scripts/engine/component.py

```diff
@@ -144,20 +145,56 @@ class Knowledge(RegisteredComponent):
     An entity's knowledge, including skills. Skills are held as skill_name : {Skill, cooldown}.
     """
 
-    def __init__(self, skills: Dict[str, Dict[str, Any]] = None, skill_order: List[str] = None):
-        if skills is None:
-            skills = {}
-        if skill_order is None:
-            skill_order = []
-
-        self.skill_order = skill_order  # list of skill names, to allow access by index
-        self.skills: Dict[str, Dict[str, Any]] = skills  # skill_name : {Skill, cooldown}
-            # FIXME - how is it str and Skill? Can't be both
```

Comment:
I felt it was better to add a couple of functions and abstract the usage from the internal implementation. Now if the internal structure of the Knowledge component changes we won't need to change anything outside of that.